### PR TITLE
draft for 1.3.1 release

### PR DESCRIPTION
**Pull request by gaktive**
still need to verify the go router workaround under Known Issues.
also tidied up incorrect info regarding postgres role from 1.3.0 release

### CODE REVIEW

**Comment by cjschroder** [comment]
:+1:  for spelling and grammar

File: adoc/Release-Notes.adoc [hunk 1]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+** ruby-buildpack: 1.7.31
+** staticfile-buildpack: 1.4.39
+
+
+[id='sec.1_3_1.issue']
+==== Known Issues
+
+* For deployments on Amazon EKS: the The {awsa} Service Broker(https://aws.amazon.com/partners/servicebroker/) should now be used instead of the deprecated `cf-brokers` wrapper.
+
+* For custom PSPs, `SYS_RESOURCE` no longer needs to specified under added capabilities in the role-manifest.yml
```
Comment (satadruroy) [inline adoc/Release-Notes.adoc]: It should be the values yaml. e.g., scf-config-values.yaml instead of role-manifest.yml?

File: adoc/Release-Notes.adoc [hunk 2]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+
+* During an upgrade from 2.14 to 2.15.2, the GoRouter and the applications it routes to will be unavailable until the new GoRouter pods are ready. You can work around this by setting the following label on the existing GoRouter pod specs:
+labels:
+
+.. `app.kubernetes.io/component: "router"`
+.. `skiff-role-name: "router"`
+
```
Comment (satadruroy) [inline adoc/Release-Notes.adoc:121]: I think all we need (subject to confirmation from QA that this actually works) is the first label, i.e., `app.kubernetes.io/component: "router"` - the `skiff-role-name` is the old label and already exists on 2.14.

**Review by satadruroy — CHANGES_REQUESTED** [review]
Couple of minor changes requested - please see inline comments.

File: adoc/Release-Notes.adoc (hunk 1 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc]: Yup, wrong file.

File: adoc/Release-Notes.adoc (hunk 2 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc:121]: Do let me know once this is confirmed by QA.

File: adoc/Release-Notes.adoc [hunk 3]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+For information about deploying and administering {product}, see the product manuals at
+{doc-url}.
+
+
+[id='sec.1_3_1.feature']
+==== Features and Fixes
+
+* App-autoscaler no longer dependent on hairpin
+* Credhub on Azure is now supported
```
Comment (ghost) [inline adoc/Release-Notes.adoc]: `CredHub on {azure}`

File: adoc/Release-Notes.adoc [hunk 4]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+** For a full list of features and fixes see https://github.com/SUSE/stratos/releases/tag/2.3.0.
+
+For information about deploying and administering {product}, see the product manuals at
+{doc-url}.
+
+
+[id='sec.1_3_1.feature']
+==== Features and Fixes
+
+* App-autoscaler no longer dependent on hairpin
```
Comment (ghost) [inline adoc/Release-Notes.adoc]: Use correct capitalization for App-AutoScaler and use a verb: `App-AutoScaler no longer depends on hairpin`

File: adoc/Release-Notes.adoc [hunk 5]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+* Credhub on Azure is now supported
+* Corrected service name to work with `syslog` drains
+* Certificates rely on correct UAA FQDN
```
Comment (ghost) [inline adoc/Release-Notes.adoc]: I think this is hard to understand, at least expand "UAA" to "User Account and Authentication Server (UAA)"

File: adoc/Release-Notes.adoc [hunk 6]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+* Removed obsolete key and diego-cell readiness probe from role-manifest.yml
```
Comment (ghost) [inline adoc/Release-Notes.adoc]: `role-manifest.yml` is a file name, so put it within \` chars.

File: adoc/Release-Notes.adoc [hunk 7]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+** php-buildpack: 4.3.70
+** python-buildpack: 1.6.27
+** ruby-buildpack: 1.7.31
+** staticfile-buildpack: 1.4.39
+
+
+[id='sec.1_3_1.issue']
+==== Known Issues
+
+* For deployments on Amazon EKS: the The {awsa} Service Broker(https://aws.amazon.com/partners/servicebroker/) should now be used instead of the deprecated `cf-brokers` wrapper.
```
Comment (ghost) [inline adoc/Release-Notes.adoc]: * `Amazon EKS` -> `{eksa}`
* `the The` -> `The`

File: adoc/Release-Notes.adoc [hunk 8]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+
+* For custom PSPs, `SYS_RESOURCE` no longer needs to specified under added capabilities in the scf-config-values.yml
```
Comment (ghost) [inline adoc/Release-Notes.adoc]: * `no longer needs to specified` -> `no longer needs to be specified`
* put `scf-config-values.yml` in \` chars

File: adoc/Release-Notes.adoc [hunk 9]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+
+* During an upgrade from 2.14 to 2.15.2, the GoRouter and the applications it routes to will be unavailable until the new GoRouter pods are ready. You can work around this by setting the following label on the existing GoRouter pod specs:
+labels:
```
Comment (ghost) [inline adoc/Release-Notes.adoc:113]: Is this a code block?
If so, put it in code block markers:

```
+
[source]
----
labels:
.. `app.kubernetes.io/component: "router"`
.. `skiff-role-name: "router"`
----
```

If it is not a code block, the `labels:` text looks a bit misplaced.

File: adoc/Release-Notes.adoc [hunk 10]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+
+.. `app.kubernetes.io/component: "router"`
+.. `skiff-role-name: "router"`
+
+
+* The App Autoscaler services are exposed as Kube services of type LoadBalancer with public IP addresses/hostnames. However, no separate DNS entries need to be created for those addresses/hostnames as calls to autoscaler will be routed through the GoRouter.
```
Comment (ghost) [inline adoc/Release-Notes.adoc]: * `App Autoscaler` -> `App-AutoScaler`
* Second sentence is a bit clumsy. Maybe rephrase to: `Calls to App-AutoScaler are routed through the GoRouter. Therefore, you do not need to create separate DNS entries for those addresses/hostnames.`

File: adoc/Release-Notes.adoc [hunk 11]
```diff
@@ -31,6 +31,124 @@ endif::[]
...
+
+* Minibroker with {mariadb} will see timeout issues upon deletion. If an error appears, wait one minute and retry. If the `cf delete-service` command fails but the instance pods are removed from {k8s}, the service instance data can safely be removed with a `cf purge-service-instance` command.
```
Comment (ghost) [inline adoc/Release-Notes.adoc]: I know this is copied content but I _still_ wonder what "will see timeout issues" actually means -- will the Minibroker have a timeout? Will the database have a timeout? Something else? Also, what are we deleting, the database or the minibroker?

**Review by ghost — CHANGES_REQUESTED** [review]
On the face of it, it does look pretty good, minor issues noted.

However, I continue to grapple with these release notes as a whole because they really are hard to understand --

* acronyms everywhere
* very short, seemingly developer-focused feature descriptions
* instructions that are so short, they will only make sense if you are already know the product very well.

File: adoc/Release-Notes.adoc (hunk 3 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc]: Fixed

File: adoc/Release-Notes.adoc (hunk 6 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc]: Fixed.

File: adoc/Release-Notes.adoc (hunk 5 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc]: UAA already exists throughout the release notes as its initials since that's the upstream convention. I'll reword this to "Certificates rely on correct FQDN for UAA" for now and discuss with product (who still has to review)

File: adoc/Release-Notes.adoc (hunk 7 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc]: Fixed.

File: adoc/Release-Notes.adoc (hunk 8 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc]: Done.

File: adoc/Release-Notes.adoc (hunk 9 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc:113]: An initial attempt at formatting this rendered the rest of the doc mangled, hence the simplified initial draft. Thanks!

File: adoc/Release-Notes.adoc (hunk 10 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc]: Makes sense.

File: adoc/Release-Notes.adoc (hunk 11 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc]: Let me get clarification here again.

Comment (troytop) [reply adoc/Release-Notes.adoc]: Suggest: Deletion of {mariadb} instances created with Minibroker can fail with timeouts.

File: adoc/Release-Notes.adoc (hunk 10 above)
Comment (troytop) [reply adoc/Release-Notes.adoc]: Agree with @sknorr about the wording of the second sentence. Additionally, we might want to advise people that the load balancer instances can be manually deleted (need to verify). ELBs cost $0.025 per hour at idle, so it doesn't make sense to have these running if they are not used.

File: adoc/Release-Notes.adoc (hunk 11 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc]: Updated.

File: adoc/Release-Notes.adoc (hunk 2 above)
Comment (gaktive) [reply adoc/Release-Notes.adoc:121]: We can revise this section later once we confirm if one step is unnecessary.

**Review by gaktive — COMMENTED** [review]
Per the overall concern about how technical these release notes are, they're not designed to be read first. All communication regarding CAP points to going through the Deployment Guide first, then at which point, the Release Notes are invoked for breaking changes. We may re-evaluate that later but for right now, the release notes aren't the entry point. It's intricate software, to say the least.

File: adoc/Release-Notes.adoc [hunk 12]
```diff
@@ -31,6 +31,127 @@ endif::[]
...
+labels:
+.. `app.kubernetes.io/component: "router"`
+.. `skiff-role-name: "router"`
+----
+
+* The App-AutoScaler services are exposed as Kube services of type LoadBalancer with public IP addresses/hostnames. Calls to App-AutoScaler are routed through the GoRouter. Therefore, you do not need to create separate DNS entries for those addresses/hostnames.
+
+* Deletion of {mariadb} instances created with Minibroker can fail with timeouts. If an error appears, wait one minute and retry. If the `cf delete-service` command fails but the instance pods are removed from {k8s}, the service instance data can safely be removed with a `cf purge-service-instance` command.
+
+* On {azure} it is recommended to run on instance types DS4_v2 or larger due to the introduction of the cflinuxfs3 stack.
```
Comment (HartS) [inline adoc/Release-Notes.adoc:126]: I think this should note the instance type `Standard_DS4_v2` and also mention the premium storageclass

File: adoc/Release-Notes.adoc [hunk 13]
```diff
@@ -60,7 +181,7 @@ For information about deploying and administering {product}, see the product man
 ** `api-group` replaces `api`
 * UAA charts now have affinity/antiaffinity logic
 * Exposed SMTP_HOST & SMTP_FROM_ADDRESS variables to allow for account creation & password reset
-* `consul` and `postgres` roles removed as they had become redundant
+* `consul` role removed due to redundandcy
```
Comment (HartS) [inline adoc/Release-Notes.adoc:184]: There's a redundancy in this spelling of `redundancy`

**Comment by cjschroder** [comment]
> Per the overall concern about how technical these release notes are, they're not designed to be read first. All communication regarding CAP points to going through the Deployment Guide first, then at which point, the Release Notes are invoked for breaking changes. We may re-evaluate that later but for right now, the release notes aren't the entry point. It's intricate software, to say the least.

I appreciate Stefan's detailed review. He is right: however the release notes are used, they need to have clarity and coherence, and the other points he mentioned.

**Comment by ghost** [comment]
> Per the overall concern about how technical these release notes are, they're not designed to be read first.

It's not about how technical they are -- by their nature, release notes need to be very technical. But it seems to me current CAP release notes are too condensed to understand for a large class of people and this unnecessarily limits their target audience.
(Please don't interpret this as me just trying to make the CAP release notes look bad -- they're still in better shape and more complete than those of a few other products. Essentially, I am able to critique because there is something to work with here. :) )

Whether you _intend_ that people see the release notes first or not is probably not much of a deciding factor, for a significant number of people the release notes _will_ be their first foray into the product, if just to get an impression of the product and its progress (think potential customers, journalists, ...).


> All communication regarding CAP points to going through the Deployment Guide first, then at which point, the Release Notes are invoked for breaking changes.

True -- if you're installing or upgrading, the main instructions are in the Deployment Guide and that is where they belong as well. 
However, the breaking changes documented in the release notes are essentially just mentioned with the reader having to figure out all the actual steps of dealing with those changes by themselves. (Not true in _all_ cases, but in many.)


> It's intricate software, to say the least.

Agreed, but I think that is also true for other SUSE products. Yet, the reason people pay SUSE is that we promise to make stuff easier for them. Otherwise they could just go with the upstream project.
